### PR TITLE
symphony: accept caller-selected run ids

### LINKED ISSUE #2779: symphony: accept cleanup-safe caller-supplied run IDs
## Problem

`POST /api/v1/runs` allocates a run id inside `Runtime.Ingress` and returns it only after the run starts. If the server creates the run but the response is dropped, times out, or cannot be persisted by the caller, the caller cannot identify the run to cancel it. A postmerge verifier can therefore leak a live Codex room even when its cleanup runs in `finally`.

Repro:

1. POST a manual workflow start.
2. Let `Ingress.start_by_name/3` create the run.
3. Drop the HTTP response before the client receives `run_ids`.
4. Observe that the client has no deterministic id for `POST /api/v1/ir/runs/:id/cancel`.

## Desired contract

Allow a manual start to carry a caller-supplied run id through the API to `Ingress.start_by_name/3`. Reuse the existing run-id validation and ownership boundary. Define one exact replay rule based on current Runtime/Store ownership:

- an identical replay is idempotent and returns the existing run, or
- every duplicate is an explicit conflict before any second run starts.

A mismatched replay must never start or adopt another run. Add API tests for initial create, identical replay, mismatched or duplicate ownership, and cancellation by the preselected id after a simulated lost response.

This must land before ix's trusted agent-toolchain rollout uses the API.

(sent by an AI agent via Codex)

### PR DESCRIPTION
**Pull request by andrewgazelka**
## Summary

- accept and centrally validate caller-selected `run_id` values on `POST /api/v1/runs`
- atomically grant durable ownership before a run starts, with every duplicate returning 409 without overwrite
- cover create, replay, mismatch, invalid input, concurrent ownership, and cancellation after a dropped response

## Validation

- `nix build .#checks.x86_64-linux.symphony-elixir`
- 460 tests, strict Credo clean
- `nix run .#lint` remains red only on unrelated current-main findings tracked by #2460

Fixes #2779

(sent by an AI agent via Codex)

> [!NOTE]
> ### Add caller-selected run IDs with idempotent conflict handling to Symphony
> - `POST /api/v1/runs` now accepts an optional `run_id` for named workflow starts; supplying `run_id` without a workflow name returns 422.
> - `Ingress.start_workflow` validates caller-supplied IDs against a shared grammar (1–128 bytes, lowercase alphanumeric with interior hyphens) or generates a conforming ID if none is provided.
> - `Store.create/2` atomically claims durable ownership of a run ID via a temp file + hard link; concurrent creates return `{:error, :already_exists}` to the loser.
> - Both the API controller and IR run controller return HTTP 409 when a run ID conflict is detected, enabling safe replay of lost create responses.
> - Risk: runs with a caller-supplied `run_id` now fail at startup if the ID is already claimed on disk, rather than proceeding with a duplicate.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 7a99002.</sup>
> 
>

### CODE REVIEW

**Comment by andrewgazelka** [comment]
CI reached the known current-main blocker in run https://github.com/indexable-inc/index/actions/runs/29137693064:

- `rust-mc-protocol.clippy` fails at `packages/minecraft/protocol/src/slp.rs:88` on `clippy::anonymous_tuple_return_type`
- the baseline fix is PR #2753
- issue #2761 is the next known baseline blocker after those clippy diagnostics are cleared

I cancelled the remaining CVE regression run to release scarce runner capacity. I will not rerun blindly; after #2753 and #2761 land, I will rebase, rerun CI, and own this PR through merge.

(sent by an AI agent via Codex)

**Comment by andrewgazelka** [comment]
I cancelled queued required runs 29140475672 and 29140475718. ix #6992 merged its reviewed insecure head before the broker fix, so production deployment and new shared-runner admission remain held until the secure follow-up is merged, deployed, and passes the loaded canary. This PR's branch content is unchanged; both required workflows will be rerun after that release.

(sent by an AI agent via Codex)

File: packages/agent/symphony/elixir/lib/symphony_elixir/ir/store.ex [hunk 1]
```diff
@@ -92,6 +92,34 @@ defmodule SymphonyElixir.IR.Store do
...
+  @spec create(RunGraph.t(), keyword()) :: :ok | {:error, term()}
+  def create(%RunGraph{} = graph, opts \\ []) do
+    store_dir = dir(opts)
+    File.mkdir_p!(store_dir)
+    path = run_path(store_dir, graph.run_id)
+    tmp = path <> ".create-#{System.unique_integer([:positive, :monotonic])}.tmp"
+
+    try do
+      with {:ok, encoded} <- Jason.encode(encode(graph), pretty: true),
+           :ok <- File.write(tmp, encoded, [:exclusive]) do
```
[the remark this answers is not in the document]
Comment (andrewgazelka) [reply packages/agent/symphony/elixir/lib/symphony_elixir/ir/store.ex:111]: Confirmed on current head `27d49e82`: the monotonic suffix can repeat after a VM restart and a stale temp then returns the wrong failure. I will use an unrepeatable temp identity and cover crash residue before merge, after the #7003 release. (sent by an AI agent via Codex)

File: packages/agent/symphony/elixir/lib/symphony_elixir_web/controllers/ir_run_controller.ex [hunk 2]
```diff
@@ -56,6 +56,9 @@ defmodule SymphonyElixirWeb.IRRunController do
       {:error, {:workflow_not_found, _}} = reason ->
         conn |> put_status(:not_found) |> json(%{error: inspect(reason)})
 
+      {:error, {:run_id_conflict, run_id}} ->
+        conn |> put_status(:conflict) |> json(%{error: "run id already exists: #{run_id}"})
```
[the remark this answers is not in the document]
Comment (andrewgazelka) [reply packages/agent/symphony/elixir/lib/symphony_elixir_web/controllers/ir_run_controller.ex:60]: Confirmed on current head `27d49e82`: the canonical IR create action still passes an empty option list, so caller `run_id` is discarded. I will thread the parsed option through this route and add create, conflict, and cancellation coverage after the #7003 release. (sent by an AI agent via Codex)
